nodejs/awsx/apigateway/api.ts

```diff
@@ -404,6 +419,30 @@ interface SwaggerOperation {
     responses?: { [code: string]: SwaggerResponse };
     "x-amazon-apigateway-integration": ApigatewayIntegration;
     "x-amazon-apigateway-request-validator"?: reqvalidation.RequestValidator;
+
+    /**
+     * security is an object whose properties are authorizerNames. Each authorizerName refers to a
+     * SecurityDefinition, defined at the top level of the swagger definition, by matching a Security
+     * Definition's name property. The authorizerNames' values are empty arrays.
+     */
+    security?: { [authorizerName: string]: string[] }[];
+}
+
+interface SecurityDefinition {
+    type: "apiKey";
+    name: string;
+    in: "header" | "query";
+    "x-amazon-apigateway-authtype": string;
+    "x-amazon-apigateway-authorizer": LambdaAuthorizer;
+}
+
+interface LambdaAuthorizer {
+    type: "token" | "request";
+    authorizerUri: pulumi.Input<string>;
+    authorizerCredentials: pulumi.Input<string>;
+    identitySource?: string;
+    identityValidationExpression?: string;
+    authorizerResultTtlInSeconds?: number;
```

Comment:
note: if you're worried about colliding with this, we can just call this SwaggerLambdaAuthorizer.

Reply:
LambdaAuthorizer is not exposed here so this shouldnt be an issue for now.